src/cdk-experimental/menu/menu-trigger.ts

```diff
@@ -8,95 +8,304 @@
+  outputs: ['opened: cdkMenuOpened', 'closed: cdkMenuClosed'],
+  providers: [
+    {provide: MENU_TRIGGER, useExisting: CdkMenuTrigger},
+    PARENT_OR_NEW_MENU_STACK_PROVIDER,
+  ],
 })
-export abstract class MenuTrigger implements OnDestroy {
-  /** A list of preferred menu positions to be used when constructing the `FlexibleConnectedPositionStrategy` for this trigger's menu. */
-  menuPosition: ConnectedPosition[];
+export class CdkMenuTrigger extends CdkMenuTriggerBase implements OnDestroy {
+  constructor(
+    /** The DI injector for this component. */
+    injector: Injector,
+    /** The host element. */
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    /** The view container ref for this component. */
+    viewContainerRef: ViewContainerRef,
+    /** The CDK overlay service. */
+    private readonly _overlay: Overlay,
+    /** The Angular zone. */
+    private readonly _ngZone: NgZone,
+    /** The menu stack this trigger belongs to. */
+    @Inject(MENU_STACK) menuStack: MenuStack,
+    /** The parent menu this trigger belongs to. */
+    @Optional() @Inject(CDK_MENU) private readonly _parentMenu?: Menu,
+    /** The menu aim service used by this menu. */
+    @Optional() @Inject(MENU_AIM) private readonly _menuAim?: MenuAim,
+    /** The directionality of the page. */
+    @Optional() private readonly _directionality?: Directionality,
+  ) {
+    super(injector, viewContainerRef, menuStack);
+    this._registerCloseHandler();
+    this._subscribeToMenuStackClosed();
+    this._subscribeToMouseEnter();
+    this._subscribeToMenuStackHasFocus();
+  }
+
+  /** Toggle the attached menu. */
+  toggle() {
+    this.isOpen() ? this.close() : this.open();
+  }
+
+  /** Open the attached menu. */
+  open() {
+    if (!this.isOpen()) {
+      this.opened.next();
+
+      this.overlayRef = this.overlayRef || this._overlay.create(this._getOverlayConfig());
+      this.overlayRef.attach(this.getMenuContentPortal());
+      this._subscribeToOutsideClicks();
+    }
+  }
+
+  /** Close the opened menu. */
+  close() {
+    if (this.isOpen()) {
+      this.closed.next();
+
+      this.overlayRef!.detach();
+    }
+    this._closeSiblingTriggers();
+  }
+
+  /**
+   * Get a reference to the rendered Menu if the Menu is open and rendered in the DOM.
+   */
+  getMenu(): Menu | undefined {
+    return this.childMenu;
+  }
 
-  /** Emits when the attached menu is requested to open */
-  readonly opened: EventEmitter<void> = new EventEmitter();
+  /**
+   * Handles keyboard events for the menu item.
+   * @param event The keyboard event to handle
+   */
+  _toggleOnKeydown(event: KeyboardEvent) {
+    const isParentVertical = this._parentMenu?.orientation === 'vertical';
+    const keyCode = event.keyCode;
+    switch (keyCode) {
+      case SPACE:
+      case ENTER:
+        event.preventDefault();
+        this.toggle();
+        this.childMenu?.focusFirstItem('keyboard');
```

Comment:
I think the current menu doesn't do anything on space/enter if a modifier key is pressed.

Reply:
yep, good catch